### PR TITLE
Add rspec and gems for test

### PR DESCRIPTION
**Pull request by diliaaragon**
- Add Rspec at project

- Add Database_cleaner gem and its config

- Add Shoulda-Matchers gem and its configurations

- Add Factory_bot_rails gem and its config

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,45 +1,32 @@
...
-
-# Use Capistrano for deployment
-# gem 'capistrano-rails', group: :development
-
-# Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.1.0', require: false
 
-# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
-# gem 'rack-cors'
+gem 'faker'
```
Comment (juankhazbun) [inline Gemfile]: La gema Faker va en group: :test

File: Gemfile [hunk 2]
```diff
@@ -1,45 +1,32 @@
...
+gem 'puma', '~> 3.11'
+gem 'rails', '~> 5.2.3'
+gem 'sqlite3'
+gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 group :development, :test do
-  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'factory_bot_rails', '~> 5.0.2'
```
Comment (juankhazbun) [inline Gemfile]: esta gema solo debe estar en goup :test

File: Gemfile [hunk 3]
```diff
@@ -1,45 +1,32 @@
...
+  gem 'rspec-rails', '~> 3.8.2'
```
Comment (juankhazbun) [inline Gemfile]: Esta gema solo debe estar en group :test

Comment (diliaaragon) [reply Gemfile]: listo

File: Gemfile (hunk 1 above)
Comment (diliaaragon) [reply Gemfile]: listo

**Pull request merged**